source/include/core_sntp_client.h

```diff
@@ -0,0 +1,463 @@
+     */
+    SntpTimestamp_t lastRequestTime;
+
+    /**
+     * @brief State variable for storing the size of the SNTP packet that includes
+     * both #SNTP_PACKET_BASE_SIZE bytes plus any authentication data, if a security
+     * mechanism is used.
+     * This state variable is used for expecting the same size for an SNTP response
+     * from the server.
+     */
+    size_t sntpPacketSize;
+} SntpContext_t;
+
+/**
+ * @brief Initializes a context for SNTP client communication with SNTP/NTP
+ * servers.
+ *
+ * @param[out] pContext The user-supplied memory for the context that will be
+ * initialized to represent an SNTP client.
+ * @param[in] pTimeServers The list of decreasing order of priority of time
+ * servers that should be used by the SNTP client.
+ * @param[in] pNetworkBuffer The user-supplied memory that will be used for
+ * storing network data for SNTP client-server communication. The buffer
+ * MUST stay in scope for all the time the context is in use.
+ * @param[in] bufferSize The size of the passed buffer @p pNetworkBuffer. The buffer
+ * SHOULD be appropriately sized for storing an entire SNTP packet which includes
+ * both #SNTP_PACKET_BASE_SIZE bytes of standard SNTP packet size, and space for
+ * authentication data, if security mechanism is used to communicate with any of
+ * the time servers configured for use.
+ * @param[in] resolveDnsFunc The user-defined function for DNS resolution of time
+ * server.
+ * @param[in] getSystemTimeFunc The user-defined function for querying system
+ * time.
+ * @param[in] setSystemTimeFunc The user-defined function for correcting system
+ * time for every successful time response received from a server.
+ * @param[in] pUdpTransportIntf The user-defined function for performing network
+ * send/recv operations over UDP.
+ * @param[in] pAuthIntf (Optional) The user-defined interface for generating
+ * client authentication in SNTP requests and authenticating servers in SNTP
+ * responses, if security mechanism is used in SNTP communication with server.
+ *
+ * @return This function returns one of the following:
+ * - #SntpSuccess if the context is initialized.
+ * - #SntpErrorBadParameter if any of the passed parameters in invalid.
+ * - #SntpErrorBufferTooSmall if the buffer does not have the minimum size
+ * required for a valid SNTP response packet.
+ */
+/* @[define_sntp_init] */
+SntpStatus_t Sntp_Init( SntpContext_t * pContext,
+                        const SntpServerInfo_t * pTimeServers,
+                        size_t numOfServers,
+                        uint8_t * pNetworkBuffer,
+                        size_t bufferSize,
+                        SntpResolveDns_t resolveDnsFunc,
+                        SntpGetTime_t getSystemTimeFunc,
+                        SntpSetTime_t setSystemTimeFunc,
+                        const UdpTransportIntf_t * pTransportIntf,
+                        const SntpAuthenticationIntf_t * pAuthIntf );
```

Comment:
I'm not sure we want to abbreviate Interface to Int since it is not immediately obvious and might pertain to the datatype. I would just spell it out.

Reply:
Okay, have renamed the type to use the full word, **Interface**.